src/domain/community/user/user.service.ts

```diff
@@ -475,8 +476,21 @@ export class UserService {
     return user;
   }
 
-  async getUsers(): Promise<IUser[]> {
-    return (await this.userRepository.find({ serviceProfile: false })) || [];
+  async getUsers(limit?: number, randomSelection = false): Promise<IUser[]> {
+    const result = await this.userRepository.find({ serviceProfile: false });
+    if (!result) return [];
+    if (!limit || result.length < limit) return result;
```

Comment:
not taking in account the `randomSelection` param again

Reply:
see above